### PR TITLE
feat(Patch3):Get Task Details and Update the data in database based on field name and field value

### PR DESCRIPTION
**Pull request by RohitVarma001**
### Functionality:
Patch 3

### Solution:
Get the field name and field value from the user to get task details based on it and update the fields with new values in database

### Risk level:
- [ ] high 
- [x] medium
- [ ] low

### How to test:
1. `firebase emulators:start`
2.  `ng serve`

### CODE REVIEW

**Review by viv9k — APPROVED** [review]
Ship it

**Comment by RohitVarma001** [comment]
I have made changes Patch 3 API Please check then Merge

File: functions/patch3API.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        const oldFieldValue = request.body.data.FieldValue;
+        const newField = request.body.data.NewField;
+        const newFieldValue = request.body.data.NewFieldValue;
+        let taskId;
+        
+        const promise1 = db.collection("Organizations").doc(OrgDomain).collection("Tasks").where(oldField, "==", oldFieldValue).get().then((task) => {
+            task.forEach((doc) => {
+                taskId = doc.data().Id;
+                console.log("Executing Promise2 of Patch3");
+        desiredField = {};
```
Comment (viv9k) [inline functions/patch3API.js:22]: remain this variable as data, as data is the most appropriate word.

File: functions/patch3API.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+const db = admin.firestore();
+
+exports.patch3 = functions.https.onRequest((request, response) =>{
+    cors(request, response, () =>
+    {
+        const OrgDomain = request.body.data.OrgDomain;
+        const oldField = request.body.data.FieldName;
+        const oldFieldValue = request.body.data.FieldValue;
+        const newField = request.body.data.NewField;
+        const newFieldValue = request.body.data.NewFieldValue;
```
Comment (viv9k) [inline functions/patch3API.js:15]: The field name for new and old field is same i believe, so we don't need two different field names.
Let have only one field name, with old value and new value.

File: functions/patch3API.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        let taskId;
+        
+        const promise1 = db.collection("Organizations").doc(OrgDomain).collection("Tasks").where(oldField, "==", oldFieldValue).get().then((task) => {
+            task.forEach((doc) => {
+                taskId = doc.data().Id;
+                console.log("Executing Promise2 of Patch3");
+        desiredField = {};
+        desiredField[newField]=newFieldValue;
```
Comment (viv9k) [inline functions/patch3API.js:23]: This is normally a key value, json works in that way. So please check if we can improve this representation.

**Review by viv9k — CHANGES_REQUESTED** [review]
Some major changes requested, the idea is really good and this is now going to get into R&D code. Good job. But please complete the required changes.

**Review by viv9k — APPROVED** [review]
Ship it

**Pull request merged**